--- logwatch-7.3.6/scripts/services/secure	2013-06-13 14:28:17.742938051 +0200
+++ logwatch-7.3.6/scripts/services/secure.new	2013-06-13 14:31:12.812039746 +0200
@@ -328,6 +328,8 @@ while (defined($ThisLine = <STDIN>)) {
       $Error{$Service}{$Err}++;
    } elsif ( $ThisLine =~ /^login(\[\d+\])*: ROOT LOGIN\s+(ON|on)\s+`?tty[0-9]+/) {
       $RootLoginTTY++
+   } elsif ( $ThisLine =~ /^login(\[\d+\])*: ROOT LOGIN\s+(ON|on)\s+`?xvc[0-9]+/) {
+      $RootLoginXVC++
    } elsif ( $ThisLine =~ /^com.apple.SecurityServer: authinternal authenticated user root .*/) {
       $RootLoginTTY++
    } elsif ( (undef,$User) = ($ThisLine =~ /^login: LOGIN ON (tty|pts\/)[0-9]+ BY ([^ ]+)/ )) {
@@ -625,6 +627,10 @@ if ($RootLoginTTY) {
    print "\nRoot logins on tty\'s: $RootLoginTTY Time(s).\n";
 }
 
+if ($RootLoginXVC) {
+   print "\nRoot logins on xvcs: $RootLoginXVC Time(s).\n";
+}
+
 if (keys %UserLogin) {
    print "\nUser Login's:\n";
    foreach $User (sort {$a cmp $b} keys %UserLogin) {
