--- logwatch-7.3.6/scripts/services/postfix.orig	2013-01-14 14:35:06.000000000 +0100
+++ logwatch-7.3.6/scripts/services/postfix	2013-01-14 14:56:53.056972806 +0100
@@ -504,6 +504,7 @@
    [ 'HostnameVerification',        "d", "Hostname verification errors" ],
    [ 'HostnameValidationError',     "d", "Hostname validation error" ],
    [ 'Deliverable',                 "d", "Address is deliverable (sendmail -bv)" ],
+   [ 'SmtpProtocolViolation',       "d", 'SMTP protocol violations' ],
    [ 'Undeliverable',               "d", "Address is undeliverable (sendmail -bv)" ],
    [ 'TableChanged',                "d", "Restarts due to lookup table change" ],
    [ 'PixWorkaround',               "d", "Enabled PIX workaround" ],
@@ -1666,6 +1667,15 @@
       $Counts{'WarnConfigError'}{$key}++;
    }
 
+   # SMTP/ESMTP protocol violations
+   elsif ($p1 =~ /^(improper command pipelining) (after \S+) from ([^[]*)\[([^]]+)\](?::\d+)?/) {
+      # ProtocolViolation
+      #TDsd postfix/smtpd[24928]: improper command pipelining after RCPT from unknown[192.168.0.1]
+      my ($error,$stage,$host,$hostip) = ($1,$2,$3,$4);
+      $Totals{'SmtpProtocolViolation'}++;
+      $Counts{'SmtpProtocolViolation'}{$error}{$stage}{formathost($hostip,$host)}++;
+   }
+
    # rare messages (mostly debug) hit less frequently - keep far down the if-elsif chain
    # be sure anything placed here would not match any cases above
    elsif (( $p1 =~ /^statistics:/ )
