--- logwatch-7.3.6/scripts/services/zz-disk_space	2015-11-18 15:03:12.271040764 +0100
+++ logwatch-7.3.6-new/scripts/services/zz-disk_space	2015-11-18 15:03:01.594047694 +0100
@@ -163,17 +163,27 @@ sub DiskSpace ()
 sub DiskFull
 {
    my $o = `$disk_cmd`;
+   my @good_rows = ();
    my @rows = split('\n', $o);
-   foreach my $row (@rows)
-   {
-     my @fields = split(' ', $row);
-     my $use = $fields[4];
-     $use =~ s/%//;
-     if (($use > $diskfull_threshhold) && ($fields[0] !~ /\/dev\/scd/ ) && ($fields[0] !~ /\/dev\/sr/ )
-        && ($fields[0] !~ /\/dev\/loop./))
-     {
-       print "$fields[0] => $fields[4] Used. Warning. Disk Filling up.\n";
-     }
+   for (my $i=0; $i<scalar(@rows); $i++) {
+      my $row = $rows[$i];
+      my @fields = split(' ', $row);
+      my $filesystem = $fields[0];
+      my $use = -1;
+      if (scalar(@fields) == 1) {
+         # Combine with the next row
+         my @new_fields = split(' ', $rows[$i+1]);
+         $use = $new_fields[4-1];
+         $i++;
+      } else {
+         $use = $fields[4];
+      }
+      $use =~ s/%//;
+      if (($use > $diskfull_threshhold) && ($filesystem !~ /\/dev\/scd/ ) && ($filesystem !~ /\/dev\/sr/ )
+         && ($filesystem !~ /\/dev\/loop./))
+      {
+         print "$filesystem => $use% Used. Warning. Disk Filling up.\n";
+      }
    }
 }
 
