--- logwatch-7.3.6/scripts/services/sshd	2013-06-13 14:23:35.446779488 +0200
+++ logwatch-7.3.6/scripts/services/sshd.new	2013-06-13 14:23:03.000000000 +0200
@@ -169,6 +169,7 @@ my %MisMatch = ();
 my %KrbAutFail = ();
 my %KrbAutErr = ();
 my %KrbErr = ();
+my %KrbRealm = ();
 my @BadRSA = ();
 my @Scanned = ();
 my %OtherList = ();
@@ -358,6 +359,8 @@ while (defined(my $ThisLine = <STDIN>))
       $KrbErr{"error removing file " . $Error}++;     
    } elsif ( my ($Pom,$Error) = ($ThisLine =~ /pam_krb5(\[\d*\]): error resolving user name '[^ ]*' to uid\/gid pai/)) {
       $KrbErr{"error resolving user name '$Error' to uid\/gid pai"}++;
+  } elsif (my ($user,$realm) = ($ThisLine =~ /Authorized to ([^ ]+), krb5 principal \1@([^ ]+) \(krb5_kuserok\)/)) {
+      $KrbRealm{$realm}{$user}++;
       # error: chmod /dev/pts/0 0666 failed: Permission denied
    } elsif (my ($File,$Perm,$Why) = ($ThisLine =~ /error: chmod (.*) (.*) failed: (.*)/)) {
       $ChmodErr{"$File,$Perm,$Why"}++;
@@ -573,6 +576,19 @@ if (keys %KrbErr) {
    }
 }
 
+if ( ($Detail == 7 && keys %KrbRealm > 1) || ($Detail > 8 && keys %KrbRealm) ){
+   print "\nSucessfull Kerberos Authentication from ",(scalar keys %KrbRealm)," realm:\n";
+   foreach my $realm (keys %KrbRealm) {
+      if($Detail > 9){
+         print "   ",$realm,":\n";
+         foreach my $user(keys %{$KrbRealm{$realm}}){
+            print "     ",$user,": ". $KrbRealm{$realm}{$user} . " Times(s)\n";
+         }
+      }else{
+         print "   ",$realm,": ". (scalar keys %{$KrbRealm{$realm}}) . " User(s)\n";
+      }
+   }
+}
 
 if (keys %DisconnectReceived) {
    print "\n\nReceived disconnect:\n";
