--- logwatch/scripts/services/up2date.orig	2009-12-16 16:47:04.000000000 +0000
+++ logwatch/scripts/services/up2date	2009-12-16 17:03:31.000000000 +0000
@@ -25,6 +25,7 @@
       $DebugCounter++;
    }
    if ( ( $ThisLine =~ /^updating login info$/ ) or
+        ( $ThisLine =~ /^updateLoginInfo\(\) login info$/ ) or
         ( $ThisLine =~ /^Opening rpmdb in \/var\/lib\/rpm\/ with option .$/ ) or
         ( $ThisLine =~ /^successfully retrieved authentication token from up2date server$/ ) or 
         ( $ThisLine =~ /^(getA|a)vailablePackageList from network$/ ) or
