--- logwatch/scripts/services/smartd	2011-02-28 12:08:10.000000000 +0100
+++ logwatch/scripts/services/smartd.new	2013-06-06 13:55:44.835000202 +0200
@@ -67,12 +67,22 @@
        # ignore
    } elsif ( ($Device) = ($ThisLine =~ /^Device: ([^,]+), is in STANDBY mode, skipping checks/ )) {
        # ignore
+   } elsif ( ($Device) = ($ThisLine =~ /^Device: ([^,]+), self-test in progress, [0-9]+% remaining/ )) {
+       # ignore
+   } elsif ( ($Device) = ($ThisLine =~ /^Device: ([^,]+), previous self-test completed without error/ )) {
+       # ignore
+   } elsif ( ($Device) = ($ThisLine =~ /^Device: ([^,]+), type changed from \'\w+\' to \'\w+\'/ )) {
+       # ignore
+   } elsif ( ($Device) = ($ThisLine =~ /^Device: ([^,]+), state (?:read from|written to)/ )) {
+       # ignore
    } elsif ( $ThisLine =~ /^file \/var\/run\/smartd.pid written containing PID [0-9]+/ ) {
        # ignore
    } elsif ( ($Device,$Msg) = ($ThisLine =~ /^ *$/ )) {
        # ignore empty lines
    } elsif ( ($ThisLine =~ /^smartd version/)
           || ($ThisLine =~ /^Home page/)
+          || ($ThisLine =~ /^.*Copyright \(C\) [0-9-]+ by Bruce Allen/)
+          || ($ThisLine =~ /^smartd [0-9\.]+ [0-9-]+/)
           || ($ThisLine =~ /configuration file/i)
           || ($ThisLine =~ /\[trip Temperature is \d+ Celsius\]/)
           || ($ThisLine =~ /^Monitoring/)
