--- a/scripts/logwatch.pl	2013-06-13 15:43:57.000000000 +0200
+++ b/scripts/logwatch.pl	2016-01-04 13:52:06.427999981 +0100
@@ -157,6 +157,7 @@
 
 sub getInt {
    my $word = shift;
+   unless (defined($word)) { return $word; }
    my $tmpWord = lc $word;
    $tmpWord =~ s/\W//g;
    return $wordsToInts{$tmpWord} if (defined $wordsToInts{$tmpWord});
@@ -165,10 +166,12 @@
    }
    return $word;
 }
-              
+
 sub CleanVars {
    foreach (keys %Config) {
-      $Config{$_} = getInt($Config{$_});
+      unless (defined $Config{$_} and ($_ eq "hostname" or $_ eq "logdir")) {
+         $Config{$_} = getInt($Config{$_});
+      }
    }
 }
 
