--- logwatch/scripts/logwatch.pl.orig	2011-02-28 12:08:10.000000000 +0100
+++ logwatch/scripts/logwatch.pl	2012-08-14 08:53:22.127027345 +0200
@@ -416,7 +416,13 @@ if (($Config{'save'} ne "") && (-e "$Con
 }
 
 if ($Config{'mailto'} eq "") {
-   $Config{'print'} = 1;
+   #Additionally check for 'save' if 'mailto' is empty
+   if ($Config{'save'} ne "") {
+      $Config{'print'} = 0;
+   }
+   else {
+      $Config{'print'} = 1;
+   }
 }
 
 if ($Config{'debug'} > 8) {
