--- logwatch-7.3.6/scripts/services/named.orig	2013-01-14 11:24:34.000000000 +0100
+++ logwatch-7.3.6/scripts/services/named	2013-01-14 11:32:00.027659927 +0100
@@ -190,6 +190,7 @@
       ($ThisLine =~ /using default UDP\/IPv[46] port range: \[[0-9]*, [0-9]*\]/) or
       ($ThisLine =~ /adjusted limit on open files from [0-9]* to [0-9]*/) or
       ($ThisLine =~ /using up to [0-9]* sockets/) or
+      ($ThisLine =~ /validating \@0x[[:xdigit:]]+: .* no valid signature found/) or
       ($ThisLine =~ /built with/)
       # too many timeouts resolving 'ns-ext.nrt1.isc.org/AAAA' (in '.'?): disabling EDNS: 3 Time(s)
    ) {
